src/pal/src/misc/time.cpp

```diff
@@ -270,6 +275,29 @@ GetTickCount64()
         }
         retval = (ts.tv_sec * tccSecondsToMillieSeconds)+(ts.tv_nsec / tccMillieSecondsToNanoSeconds);
     }
+#elif HAVE_MACH_CLOCK_MONOTONIC
+    {
+        kern_return_t machRet;
+        clock_serv_t clock;
+        mach_timespec_t mts;
+        host_t host = mach_host_self();
+
+        if((machRet = host_get_clock_service(host, SYSTEM_CLOCK, &clock)) != KERN_SUCCESS)
```

Comment:
GetTickCount is expected to be very fast API. Acquiring/releasing the clock service for each call looks pretty expensive. Could you please do some quick performance measuments, and maybe consider caching the clock service?